guides/source/api_app.md

```diff
@@ -0,0 +1,435 @@
+    *GET* requests, and return just the headers on the way out. This
+    makes *HEAD* work reliably in all Rails APIs.
+
+While you could obviously build these up in terms of existing Rack
+middlewares, I think this list demonstrates that the default Rails
+middleware stack provides a lot of value, even if you’re “just
+generating JSON”.
+
+Handled at the ActionPack layer:
+
+-   Resourceful Routing: If you’re building a RESTful JSON API, you want
+    to be using the Rails router. Clean and conventional mapping from
+    HTTP to controllers means not having to spend time thinking about
+    how to model your API in terms of HTTP.
+-   URL Generation: The flip side of routing is URL generation. A good
+    API based on HTTP includes URLs (see [the GitHub gist
+    API](http://developer.github.com/v3/gists/) for an example).
+-   Header and Redirection Responses: *head :no\_content* and
+    *redirect\_to user\_url(current\_user)* come in handy. Sure, you
+    could manually add the response headers, but why?
+-   Caching: Rails provides page, action and fragment caching. Fragment
+    caching is especially helpful when building up a nested JSON object.
+-   Basic, Digest and Token Authentication: Rails comes with
+    out-of-the-box support for three kinds of HTTP authentication.
+-   Instrumentation: Rails 3.0 added an instrumentation API that will
+    trigger registered handlers for a variety of events, such as action
+    processing, sending a file or data, redirection, and database
+    queries. The payload of each event comes with relevant information
+    (for the action processing event, the payload includes the
+    controller, action, params, request format, request method and the
+    request’s full path).
+-   Generators: This may be passé for advanced Rails users, but it can
+    be nice to generate a resource and get your model, controller, test
+    stubs, and routes created for you in a single command.
+-   Plugins: Many third-party libraries come with support for Rails that
+    reduces or eliminates the cost of setting up and gluing together the
+    library and the web framework. This includes things like overriding
+    default generators, adding rake tasks, and honoring Rails choices
+    (like the logger and cache backend).
+
+Of course, the Rails boot process also glues together all registered
+components. For example, the Rails boot process is what uses your
+*config/database.yml* file when configuring ActiveRecord.
+
+**The short version is**: you may not have thought about which parts of
+Rails are still applicable even if you remove the view layer, but the
+answer turns out to be “most of it”.
+
+### The Basic Configuration
+
+If you’re building a Rails application that will be an API server first
+and foremost, you can start with a more limited subset of Rails and add
+in features as needed.
+
+You can generate a new api Rails app:
+
+<shell>\
+\$ rails new my\_api --api\
+</shell>
```

Comment:
A few places have formatting issues in this file not translating to md correctly.